src/main/java/com/joyent/manta/monitor/Application.java

```
@@ -132,32 +136,60 @@ public static void main(final String[] args) throws InterruptedException {
+                    client.options(client.getContext().getMantaBucketsDirectory());
+                    if ("FileMultipartUploadGetDeleteChain".equals(metricPostFix)) {
+                        String msg = "Multipart upload not supported for buckets yet."
+                                + "Check the CONFIG_FILE env variable and re-run the application";
+                        LOG.error(msg);
+                        System.exit(1);
+                    }
+                }
+
                 Histogram requestPutHistogram = Histogram.build()
-                        .name("manta_monitor_put_request_latency_seconds_"
-                                + chain.getClass().getSimpleName())
+                        .name(putRequestLatencyMetric + metricPostFix)
                         .help("Metric that gives a cumulative observation for "
-                                + "latency, in seconds, in creating a directory and "
-                                + "uploading a file")
+                                + "latency, in seconds, in creating a directory/bucket and "
+                                + "putting an object")
                         .create();
                 // Add the histogram here, to register it later below
                 requestPutHistogramsMap.put(chain.getClass().getSimpleName(),
                         requestPutHistogram);
                 ChainRunner chainRunner = new ChainRunner(chain, runner,
-                        client, UNCAUGHT_EXCEPTION_HANDLER, requestPutHistogramsMap);
+                        client, UNCAUGHT_EXCEPTION_HANDLER, requestPutHistogramsMap, testType);
                 runningChains.add(chainRunner);
             } catch (ClassNotFoundException e) {
                 LOG.error("Unable to load class: {}", runner.getChainClassName());
+            } catch (IOException e) {
+                if (e instanceof MantaClientHttpResponseException) {
+                    if (((MantaClientHttpResponseException)e)
```

Do you want to ignore the exception if this criteria is not matched? It is ok if you do want to do that intentionally. If so, you should add a comment to indicate that this is an intentional part of the application's behavior and not a mistake.

---

Yes. We want to exit out in this case. I will add appropriate comments.